drop-in/src/main/java/com/adyen/checkout/dropin/ui/stored/PreselectedStoredPaymentMethodFragment.kt

```diff
@@ -137,23 +148,43 @@ class PreselectedStoredPaymentMethodFragment : DropInBottomSheetDialogFragment()
 
     private fun observe() {
         storedPaymentViewModel.storedPaymentLiveData.observe(
-            viewLifecycleOwner,
-            {
-                when (it) {
-                    is StoredCardModel -> {
-                        binding.storedPaymentMethodItem.textViewText.text = requireActivity().getString(R.string.card_number_4digit, it.lastFour)
-                        imageLoader.load(it.imageId, binding.storedPaymentMethodItem.imageViewLogo)
-                        binding.storedPaymentMethodItem.textViewDetail.text = DateUtils.parseDateToView(it.expiryMonth, it.expiryYear)
-                        binding.storedPaymentMethodItem.textViewDetail.visibility = View.VISIBLE
-                    }
-                    is GenericStoredModel -> {
-                        binding.storedPaymentMethodItem.textViewText.text = it.name
-                        binding.storedPaymentMethodItem.textViewDetail.visibility = View.GONE
-                        imageLoader.load(it.imageId, binding.storedPaymentMethodItem.imageViewLogo)
-                    }
+            viewLifecycleOwner
+        ) {
+            binding.storedPaymentMethodItem.root.setDragLocked(!it.isRemovable)
+            when (it) {
+                is StoredCardModel -> {
+                    binding.storedPaymentMethodItem.textViewText.text =
+                        requireActivity().getString(R.string.card_number_4digit, it.lastFour)
+                    imageLoader.load(it.imageId, binding.storedPaymentMethodItem.imageViewLogo)
+                    binding.storedPaymentMethodItem.textViewDetail.text =
+                        DateUtils.parseDateToView(it.expiryMonth, it.expiryYear)
+                    binding.storedPaymentMethodItem.textViewDetail.visibility = View.VISIBLE
+                }
+                is GenericStoredModel -> {
+                    binding.storedPaymentMethodItem.textViewText.text = it.name
+                    binding.storedPaymentMethodItem.textViewDetail.visibility = View.GONE
+                    imageLoader.load(it.imageId, binding.storedPaymentMethodItem.imageViewLogo)
                 }
             }
-        )
+        }
+    }
+
+    private fun showRemoveStoredPaymentDialog() {
```

Comment:
I thought it was not gonna be possible to delete the PM in this screen. Did you check this with Arjen? Seems very counter intuitive.

Reply:
Yep checked it with Arjen and decided to do it like this